### PR TITLE
Add controller rumbling command

### LINKED ISSUE #27: Controller Vibration
Might be a useful resource: https://www.chiefdelphi.com/t/how-to-use-the-rumble-function-on-a-xbox-controller-both-360-and-one-in-java/358167

### PR DESCRIPTION
**Pull request by m10653**
Closes #27

### CODE REVIEW

**Comment by m10653** [comment]
Heads up simulating the rumble does work in sim with no additional code. However you must select use real driver station option with simulation robot code and enable the robot with the NI driver station for it to be able to control the rumble motors.

File: src/main/java/org/jmhsrobotics/frc2024/Robot.java [hunk 1]
```diff
@@ -104,10 +104,6 @@ public void autonomousInit() {
 	public void autonomousPeriodic() {
 	}
 
-	@Override
-	public void autonomousExit() {
-	}
-
```
Comment (m10653) [inline src/main/java/org/jmhsrobotics/frc2024/Robot.java:110]: Unnecessary change please remove. Is not related to controller Vibration.

File: src/main/java/org/jmhsrobotics/frc2024/Robot.java [hunk 2]
```diff
@@ -117,6 +113,7 @@ public void teleopInit() {
 
 	@Override
 	public void teleopPeriodic() {
+
```
Comment (m10653) [inline src/main/java/org/jmhsrobotics/frc2024/Robot.java:116]: Unnecessary change please remove.

File: src/main/java/org/jmhsrobotics/frc2024/RobotContainer.java [hunk 3]
```diff
@@ -73,6 +73,7 @@ private boolean getAllianceFlipState() {
 	}
 
 	private void configureBindings() {
+
```
Comment (m10653) [inline src/main/java/org/jmhsrobotics/frc2024/RobotContainer.java:76]: Unnecessary change please remove.

File: src/main/java/org/jmhsrobotics/frc2024/utils/ControllerRumble.java [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+  }
+
+  public void rumble(int duration, double strength) {
+    for (int i = 0; i < duration; i++) {
+      controller.setRumble(GenericHID.RumbleType.kLeftRumble, strength);
+      controller.setRumble(GenericHID.RumbleType.kRightRumble, strength);
+    }
+  }
+    SmartDashboard.putData(CommandScheduler.getInstance()); 
+
```
Comment (m10653) [inline src/main/java/org/jmhsrobotics/frc2024/utils/ControllerRumble.java:25]: Brackets for the method rumble to not match up with this line. The SmartDashboard.putData line is outside of any method causing a compile time error.

File: src/main/java/org/jmhsrobotics/frc2024/utils/ControllerRumble.java [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+  public ControllerRumble(GenericHID controller) {
+    this.controller = controller;
+  }
+
+  public void rumble(int duration, double strength) {
+    for (int i = 0; i < duration; i++) {
+      controller.setRumble(GenericHID.RumbleType.kLeftRumble, strength);
+      controller.setRumble(GenericHID.RumbleType.kRightRumble, strength);
+    }
```
Comment (m10653) [inline src/main/java/org/jmhsrobotics/frc2024/utils/ControllerRumble.java:22]: This for loop does not properly set the rumble for a specified time. It will just cause the thread to hold and cause loop overruns. And the speed will very based on the processing speed of the computer it is run on. You should utilize a timer instead and check if a target time as elapsed and set the rumble state to be zero after the time as elapsed.

File: src/main/java/org/jmhsrobotics/frc2024/utils/ControllerRumble.java [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+  private GenericHID controller;
+
+  public ControllerRumble(GenericHID controller) {
+    this.controller = controller;
+  }
+
+  public void rumble(int duration, double strength) {
+    for (int i = 0; i < duration; i++) {
+      controller.setRumble(GenericHID.RumbleType.kLeftRumble, strength);
+      controller.setRumble(GenericHID.RumbleType.kRightRumble, strength);
```
Comment (m10653) [inline src/main/java/org/jmhsrobotics/frc2024/utils/ControllerRumble.java:21]: There is a rumble time KBoth that you may want to use over the two lines.
Preferably the command would allow you to select what type of rumble you would like, so probably would be better to pass the Rumble Type via the constructor.

File: src/main/java/org/jmhsrobotics/frc2024/utils/ControllerRumble.java [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+import edu.wpi.first.wpilibj.XboxController;
+import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;
+import edu.wpi.first.wpilibj2.command.Command;
+import edu.wpi.first.wpilibj2.command.CommandScheduler;
+import edu.wpi.first.wpilibj2.command.Commands;
+import edu.wpi.first.wpilibj2.command.button.CommandXboxController;
+
+public class ControllerRumble {
+  private GenericHID controller;
+
```
Comment (m10653) [inline src/main/java/org/jmhsrobotics/frc2024/utils/ControllerRumble.java:13]: If you want this class to be a command. It must extend "Command"
For more information I recommend reading the following documentation
https://docs.wpilib.org/en/stable/docs/software/commandbased/index.html

File: src/main/java/org/jmhsrobotics/frc2024/utils/ControllerRumble.java [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+      controller.setRumble(GenericHID.RumbleType.kLeftRumble, strength);
+      controller.setRumble(GenericHID.RumbleType.kRightRumble, strength);
+    }
+  }
+    SmartDashboard.putData(CommandScheduler.getInstance()); 
+
+  public void disabledInit() {
+    
+    // add code to initialize rumble during disabled mode here
+  }
```
Comment (m10653) [inline src/main/java/org/jmhsrobotics/frc2024/utils/ControllerRumble.java:29]: This method is not part of the command class, and will not run when disabled. you must use 
```java
@Override
public void end(boolean interrupted) {
}
```

If you want code to run when the command ends.

**Review by Lu-han-wang — APPROVED** [review]
